src/refiners/foundationals/latent_diffusion/model.py

```diff
@@ -91,15 +91,17 @@ def forward(
         unconditional_prediction, conditional_prediction = self.unet(latents).chunk(2)
 
         # classifier-free guidance
-        noise = unconditional_prediction + condition_scale * (conditional_prediction - unconditional_prediction)
+        predicted_noise = unconditional_prediction + condition_scale * (
+            conditional_prediction - unconditional_prediction
+        )
         x = x.narrow(dim=1, start=0, length=4)  # support > 4 channels for inpainting
 
         if self.has_self_attention_guidance():
-            noise += self.compute_self_attention_guidance(
+            predicted_noise += self.compute_self_attention_guidance(
                 x=x, noise=unconditional_prediction, step=step, clip_text_embedding=clip_text_embedding, **kwargs
```

Comment:
We could have been tempted to do some renaming for `compute_self_attention_guidance` `noise` argument. But let's keep it for later (definitely fine as-is)

Reply:
Yeah true. 

Also FYI I noticed that self_attention_guidance doesn't work with the euler scheduler. Some typing mismatch that I didn't look into.

Reply:
> FYI I noticed that self_attention_guidance doesn't work with the euler scheduler

Definitely - which is also why it has to be revamped (e.g. `remove_noise` must be implemented differently in this case).